corehq/form_processor/backends/sql/processor.py

```diff
@@ -104,30 +104,42 @@ def save_processed_models(cls, processed_forms, cases=None, stock_result=None, p
                 ledger_value.db for ledger_value in stock_result.models_to_save
             }
 
-        with ExitStack() as stack:
-            for db_name in db_names:
-                stack.enter_context(transaction.atomic(db_name))
+        all_models = filter(None, (
+            list(processed_forms)
+            + (cases if cases else [])
+            + (stock_result.models_to_save if stock_result else [])
+        ))
```

Comment:
nit: this is a perfect place for `itertools.chain` (to avoid constructing a list that is not used except in case of error)
```py
all_models = filter(None, chain(
    processed_forms,
    cases or [],
    stock_result.models_to_save if stock_result else [],
))
```